### DISCUSSION TITLE
How to set selected value of Select component programmatically?

### DISCUSSION THREAD

**Discussion by xuning0** [mantinedev/mantine#3822] [Q&A] [answered]
I use ref 

```
const ref = useRef<HTMLInputElement>(null);

<Select ref={ref} ... />
```

 and try `ref.current?.setAttribute('value', 'some value')`, but it doesn't seem to work.

**Comment by Sajarin-M** [comment, marked as answer]
Can't you just make the select controlled?
[Controlled Select](https://mantine.dev/core/select/#controlled)

**Reply by xuning0** [reply]
Oh my fault. This is what I tried for the first time, but only set `onChange` and forgot to set `value`. 😭
